spatialprofilingtoolbox/cggnn/extract.py

```diff
@@ -0,0 +1,57 @@
+"""Extract information cg-gnn needs from SPT."""
+
+from typing import Any
+
+from pandas import DataFrame, concat, merge  # type: ignore
+from numpy import sort  # type: ignore
+
+from spatialprofilingtoolbox.db.feature_matrix_extractor import FeatureMatrixExtractor
+
+
+def _create_cell_df(
+    dfs_by_specimen: dict[str, DataFrame],
+    feature_names: dict[str, str],
+) -> DataFrame:
+    """Find simple and complex phenotypes, and locations and merge into a DataFrame."""
+    feature_ids_to_names = {ft_id: 'FT_' + ft_name for ft_id, ft_name in feature_names.items()}
```

Comment:
How can we distinguish simple phenotypes from complex phenotypes? Inside `cg-gnn` it distinguishes between the two because the former has a `FT_` prefix and the latter have a `PH_` prefix.

Reply:
If I recall correctly, `FeatureMatrixExtractor` doesn't provide the "composite phenotype" assignments, only the dichotomized expression matrix for the original channels of imaging ("simple phenotypes").
The reasoning was that these assignments aren't really part of the source data, they are derivative.
Still, it seems you make use of these. A little helper function would be useful to this end, perhaps: 

```py
def assign_phenotypes(expressions: DataFrame, study_name: str) -> DataFrame
```
?
This function could grab all the phenotype definitions for the study, and apply `get_mask` to each one.

Reply:
Would this be standalone, in this file, or in `FeatureMatrixExtractor`?